tests/phpunit/unit-tests/class-llms-test-order-generator.php

```diff
@@ -0,0 +1,967 @@
+		$this->assertWPErrorCodeEquals( $gen::E_ORDER_NOT_FOUND, $res );
+
+	}
+
+	/**
+	 * Test confirm() when the gateway encounters an error.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_confirm_gateway_errors() {
+
+		LLMS_Forms::instance()->install( true );
+
+		$gateway = new class() extends LLMS_Payment_Gateway {
+			public $id = 'fake-confirm-err';
+			public function handle_pending_order( $order, $plan, $person, $coupon = false ) {}
+			public function confirm_pending_order( $order ) {
+				return new WP_Error( 'gateway-err', 'Message' );
+			}
+		};
+
+		$gateway->supports['recurring_payments'] = true;
+		$this->load_payment_gateway( $gateway );
+
+		$order = new LLMS_Order( 'new' );
+		$order->set( 'payment_gateway', 'fake-confirm-err' );
+
+		$data                         = $this->get_mock_checkout_data_array();
+		$data['llms_order_key']       = $order->get( 'order_key' );
+		$data['llms_payment_gateway'] = 'fake-confirm-err';
+
+		$gen = new LLMS_Order_Generator( $data );
+		LLMS_Unit_Test_Util::set_private_property( $gen, 'gateway', $gateway );
+
+		$res = $gen->confirm();
+
+		$this->assertIsWPError( $res );
+		$this->assertWPErrorCodeEquals( 'gateway-err', $res );
+
+		$this->unload_payment_gateway( 'fake-confirm-err' );
+
+	}
+
+	/**
+	 * Test confirm() success
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_confirm_success() {
+
+		$gateway = new class() extends LLMS_Payment_Gateway {
+			public $id = 'fake-confirm-success';
+			public function handle_pending_order( $order, $plan, $person, $coupon = false ) {}
+			public function confirm_pending_order( $order ) {
+				return array(
+					'txn_id' => 12345,
+				);
+			}
+		};
+
+		$gateway->supports['recurring_payments'] = true;
+		$gateway->set_option( 'enabled', 'yes' );
+		$this->load_payment_gateway( $gateway );
+
+		$order = new LLMS_Order( 'new' );
+		$order->set( 'payment_gateway', 'fake-confirm-success' );
+
+		$data                         = $this->get_mock_checkout_data_array();
+		$data['llms_order_key']       = $order->get( 'order_key' );
+		$data['llms_payment_gateway'] = 'fake-confirm-success';
+
+		$gen = new LLMS_Order_Generator( $data );
+		LLMS_Unit_Test_Util::set_private_property( $gen, 'gateway', $gateway );
+
+		$this->assertEquals( array( 'txn_id' => 12345 ), $gen->confirm() );
+
+		// User data should have been stored.
+		$this->assertEquals( $data['email_address'], $order->get( 'billing_email' ) );
+
+		$this->unload_payment_gateway( 'fake-confirm-success' );
+
+	}
+
+	/**
+	 * Test create() when an error is encountered creating the order post.
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_create_error() {
+
+		// Forces an error to be encountered during order creation.
+		add_filter( 'wp_insert_post_empty_content', '__return_true' );
+
+		$gen = new LLMS_Order_Generator( array() );
+		$res = LLMS_Unit_Test_Util::call_method( $gen, 'create' );
```

Comment:
`LLMS_Unit_Test_Util::call_method()` can throw an exception, so methods that use it should have a `@throws ReflectionException` tag in the DocBlock, unless they catch it.